subsys/input/input_longpress.c

```diff
@@ -108,35 +107,38 @@ static int longpress_init(const struct device *dev)
 	return 0;
 }
 
+/* short_codes are now optional, if not set the assert shouldn't fail */
+#define SHORT_CODES_LEN(inst) DT_PROP_LEN_OR(inst, short_codes, DT_INST_PROP_LEN(inst, input_codes))
+
 #define INPUT_LONGPRESS_DEFINE(inst)                                                               \
-	BUILD_ASSERT(DT_INST_PROP_LEN(inst, input_codes) ==                                        \
-		     DT_INST_PROP_LEN(inst, short_codes));                                         \
-	BUILD_ASSERT(DT_INST_PROP_LEN(inst, input_codes) ==                                        \
-		     DT_INST_PROP_LEN(inst, long_codes));                                          \
+	BUILD_ASSERT(DT_INST_PROP_LEN(inst, input_codes) == DT_INST_PROP_LEN(inst, long_codes));   \
 	static void longpress_cb_##inst(struct input_event *evt)                                   \
 	{                                                                                          \
 		longpress_cb(DEVICE_DT_INST_GET(inst), evt);                                       \
 	}                                                                                          \
 	INPUT_CALLBACK_DEFINE(DEVICE_DT_GET_OR_NULL(DT_INST_PHANDLE(inst, input)),                 \
 			      longpress_cb_##inst);                                                \
 	static const uint16_t longpress_input_codes_##inst[] = DT_INST_PROP(inst, input_codes);    \
-	static const uint16_t longpress_short_codes_##inst[] = DT_INST_PROP(inst, short_codes);    \
+	__maybe_unused static const uint16_t longpress_short_codes_##inst[] =                      \
+		DT_INST_PROP_OR(inst, short_codes, {});                                            \
 	static const uint16_t longpress_long_codes_##inst[] = DT_INST_PROP(inst, long_codes);      \
 	static const struct longpress_config longpress_config_##inst = {                           \
 		.input_dev = DEVICE_DT_GET_OR_NULL(DT_INST_PHANDLE(inst, input)),                  \
 		.input_codes = longpress_input_codes_##inst,                                       \
-		.short_codes = longpress_short_codes_##inst,                                       \
+		.short_codes = (sizeof(longpress_short_codes_##inst) != 0)                         \
+				       ? longpress_short_codes_##inst                              \
+				       : NULL,                                                     \
```

Comment:
same as above

Reply:
hart to keep the style here, since the line length would be > 100 chars and clang formatter changes it to this style.